### PR TITLE
update dm docs index

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 
improve DM docs index

### What is changed and how it works?
1. update readme.md
2. update some directory

### Check List 

Tests 

 - No code

### CODE REVIEW

**Comment by amyangfei** [comment]
PTAL @GregoryIan @csuzhangxc

**Comment by csuzhangxc** [comment]
@amyangfei only update index, not directory structure, right?

File: docs/dm/zh_CN/README.md [hunk 1]
```diff
@@ -18,45 +18,46 @@ DM (Data Migration) 是一体化数据同步任务管理平台，支持全量备
...
+    - 任务管理
+        - 基础使用
+            - [dmctl 使用手册](./task-handling/dmctl-manual.md)
+            - [任务管理命令](./task-handling/task-commands.md)
+            - [上游数据库实例检查](./task-handling/check-mysql.md)
+            - [query-status 内容介绍](./task-handling/query-status.md)
+        - 配置文件
+            - [Task 配置文件介绍](./configuration/configuration.md)
+            - [task.yaml 示例](./configuration/task.yaml)
+        - 分库分表
```
Comment (csuzhangxc) [inline docs/dm/zh_CN/README.md]: I want to move these to `feature` when I re-write them, is it fine?

**Comment by csuzhangxc** [comment]
any link or index to `dm-worker-unit.md`?

File: docs/dm/zh_CN/README.md [hunk 2]
```diff
@@ -18,45 +18,46 @@ DM (Data Migration) 是一体化数据同步任务管理平台，支持全量备
...
+            - [分库分表支持方案](./shard-table/merge-solution.md)
```
Comment (IANTHEREAL) [inline docs/dm/zh_CN/README.md]: 分库分表要不要放在 `功能` 里面

**Comment by IANTHEREAL** [comment]
LGTM

**Comment by amyangfei** [comment]
> @amyangfei only update index, not directory structure, right?

@csuzhangxc `dm-worker-unit.md` is linked from `overview.md`

**Comment by amyangfei** [comment]
> only update index, not directory structure, right?

only move `maintenance/query-status.md` to `task-handling/query-status.md`

**Comment by amyangfei** [comment]
added a simple index in `overview.md`, PTAL @GregoryIan @csuzhangxc

File: docs/dm/zh_CN/README.md [hunk 3]
```diff
@@ -18,45 +18,46 @@ DM (Data Migration) 是一体化数据同步任务管理平台，支持全量备
...
+        - [扩充/缩减 DM 集群](./maintenance/scale-out.md)
+        - [部署目录结构](./maintenance/directory-structure.md)
+        - [重启 DM 组件注意事项](./maintenance/caution-for-restart-dm.md)
+        - [使用 dmctl 执行主-从切换](./maintenance/master-slave-switch.md)
+        - [DM 监控与告警](./maintenance/metrics-alert.md)
+    - 任务管理
+        - 基础使用
+            - [dmctl 使用手册](./task-handling/dmctl-manual.md)
+            - [任务管理命令](./task-handling/task-commands.md)
+            - [上游数据库实例检查](./task-handling/check-mysql.md)
```
Comment (csuzhangxc) [inline docs/dm/zh_CN/README.md]: we do not have `check-mysql.md` now?

Comment (IANTHEREAL) [reply docs/dm/zh_CN/README.md]: remove it

**Comment by csuzhangxc** [comment]
LGTM

File: docs/dm/zh_CN/README.md (hunk 3 above)
Comment (amyangfei) [reply docs/dm/zh_CN/README.md]: removed

**Pull request merged**